pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateTableViewSyncer.java

```diff
@@ -0,0 +1,186 @@
+            return;
+        }
+
+        if (isActive) {
+            return;
+        }
+
+        try {
+            long started = System.currentTimeMillis();
+
+            if (metadataStoreTableView != null) {
+                metadataStoreTableView.close();
+                metadataStoreTableView = null;
+            }
+            metadataStoreTableView = new ServiceUnitStateMetadataStoreTableViewImpl();
+            metadataStoreTableView.start(
+                    pulsar,
+                    this::syncToSystemTopic,
+                    (k, v) -> {}
+            );
+            log.info("Started MetadataStoreTableView");
+
+            if (systemTopicTableView != null) {
+                systemTopicTableView.close();
+                systemTopicTableView = null;
+            }
+            systemTopicTableView = new ServiceUnitStateTableViewImpl();
+            systemTopicTableView.start(
+                    pulsar,
+                    this::syncToMetadataStore,
+                    (k, v) -> {}
+            );
+            log.info("Started SystemTopicTableView");
+
+            Map<String, ServiceUnitStateData> merged = new HashMap<>();
+            metadataStoreTableView.entrySet().forEach(e -> merged.put(e.getKey(), e.getValue()));
+            systemTopicTableView.entrySet().forEach(e -> merged.put(e.getKey(), e.getValue()));
+
+            List<CompletableFuture<Void>> futures = new ArrayList<>();
+            var opTimeout = pulsar.getConfiguration().getMetadataStoreOperationTimeoutSeconds();
+
+            // Directly use store to sync existing items to metadataStoreTableView(otherwise, they are conflicted out)
+            var store = pulsar.getLocalMetadataStore();
+            var writer = ObjectMapperFactory.getMapper().writer();
+            for (var e : merged.entrySet()) {
+                futures.add(store.put(ServiceUnitStateMetadataStoreTableViewImpl.PATH_PREFIX + "/" + e.getKey(),
+                        writer.writeValueAsBytes(e.getValue()), Optional.empty()).thenApply(__ -> null));
+                if (futures.size() == MAX_CONCURRENT_SYNC_COUNT) {
+                    FutureUtil.waitForAll(futures).get(opTimeout, TimeUnit.SECONDS);
+                    futures.clear();
+                }
+            }
+            FutureUtil.waitForAll(futures).get(opTimeout, TimeUnit.SECONDS);
+            futures.clear();
+
+            for (var e : merged.entrySet()) {
+                futures.add(syncToSystemTopic(e.getKey(), e.getValue()));
+                if (futures.size() == MAX_CONCURRENT_SYNC_COUNT) {
+                    FutureUtil.waitForAll(futures).get(opTimeout, TimeUnit.SECONDS);
+                    futures.clear();
+                }
+            }
+            FutureUtil.waitForAll(futures).get(opTimeout, TimeUnit.SECONDS);
+            futures.clear();
+
+            int size = merged.size();
+            int syncTimeout = opTimeout * size;
```

Comment:
Is the sync timeout too big here? If `opTimeout` is `30` and the size is `10`, then the `syncTimeout` will be `300` seconds.

Reply:
sure. let me add some cap here, for 5 mins here.